tests/phpunit/includes/testcase-canonical.php

```diff
@@ -1,54 +1,86 @@
 <?php
 
-class PLL_Canonical_UnitTestCase extends PLL_UnitTestCase {
-
-	// mainly copy paste from WP_Canonical_UnitTestCase::assertCanonical
-	public function assertCanonical( $test_url, $expected ) {
-		if ( is_string( $expected ) ) {
-			$expected = array( 'url' => $expected );
-		} elseif ( is_array( $expected ) && ! isset( $expected['url'] ) && ! isset( $expected['qv'] ) ) {
-			$expected = array( 'qv' => $expected );
-		}
+class PLL_Canonical_UnitTestCase extends WP_Canonical_UnitTestCase {
+	use PLL_UnitTestCase_Trait;
 
-		if ( ! isset( $expected['url'] ) && ! isset( $expected['qv'] ) ) {
-			$this->markTestSkipped( 'No valid expected output was provided' );
-		}
+	private $options;
 
-		$this->go_to( home_url( $test_url ) );
+	public function setUp() {
+		parent::setUp();
 
-		// Does the redirect match what's expected?
-		$can_url = self::$polylang->filters_links->check_canonical_url( home_url( $test_url ), false ); // FIXME TODO define links ( need $curlang )
-		if ( $wp_can_url = redirect_canonical( $can_url, false ) ) {
-			$parsed_can_url = wp_parse_url( $wp_can_url );
-		} else {
-			$parsed_can_url = wp_parse_url( $can_url );
-		}
+		add_filter( 'wp_using_themes', '__return_true' ); // To pass the test in PLL_Choose_Lang::init() by default.
+		add_filter( 'wp_doing_ajax', '__return_false' );
 
-		// Just test the Path and Query if present
-		if ( isset( $expected['url'] ) ) {
-			$this->assertEquals( $expected['url'], $parsed_can_url['path'] . ( ! empty( $parsed_can_url['query'] ) ? '?' . $parsed_can_url['query'] : '' ) );
-		}
+		$this->options = array_merge(
+			PLL_Install::get_default_options(),
+			array(
+				'default_lang' => 'en',
+				'hide_default' => 0,
+				'post_types'   => array(
+					'cpt' => 'pllcanonical',
+					// translate the cpt // FIXME /!\ 'after_setup_theme' already fired and the list of translated post types is already cached :(
```

Comment:
Does this `FIXME` still apply?

Reply:
Yes probably. The list of translated post types is cached here https://github.com/polylang/polylang/blob/2.8.3/include/model.php#L241-L243